Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -686,7 +686,7 @@ private void CleanUpUiRaycastCamera()
 
             if (uiRaycastCamera != null)
             {
-                UnityEngine.Object.Destroy(uiRaycastCamera.gameObject);
+                UnityEngine.Object.DestroyImmediate(uiRaycastCamera.gameObject);
```

Comment:
Why do you need to use DestroyImmediate here instead of destroy? According to [this guidance](https://docs.unity3d.com/ScriptReference/Object.DestroyImmediate.html?_ga=2.78406291.626937135.1569260631-1015154058.1557790993) "This function should only be used when writing editor code since the delayed destruction will never be invoked in edit mode". Can CleanUpUiRaycastCamera run at runtime, not edit time?

Reply:
It fixed a race condition where the uiRaycastCamera was technically still present when the input system was starting back up that caused it to not get re-created.

Reply:
Can you comment that here, so people don't get surprised why you are using DestroyImmediate at runtime?

Reply:
Yeah, the documentation makes it sound really scary to use DestroyImmediate.  But if you have something where you need to be deterministic about when it occurs, then you need to use DestroyImmediate.